### PR TITLE
Warn with verify-checksums when extra files exist in wp-admin or wp-includes

### LINKED ISSUE #2283: Warn with verify-checksums when extra files exist in wp-admin or wp-includes
Because we have a manifest of the files that should exist in these directories, we can warn, but not fail, when extra files exist in `wp-admin` or `wp-includes` that aren't in the manifest.

Previously #1905

### PR DESCRIPTION
**Pull request by gilbitron**
Resolves #2283

### CODE REVIEW

File: php/commands/core.php [hunk 1]
```diff
@@ -861,13 +861,43 @@ public function verify_checksums( $args, $assoc_args ) {
...
+		foreach ( $files as $file_info ) {
+			if ( $file_info->isFile() && 'wp-config.php' !== basename( $file_info->getPathname() ) && false === strpos( $file_info->getPathname(), 'wp-content/' ) ) {
+				$core_files[] = str_replace( ABSPATH, '', $file_info->getPathname() );
+			}
+		}
+
+		return $core_files;
+	}
+
+	private function remove_wp_content_files_filter( $file ) {
```
Comment (danielbachhuber) [inline php/commands/core.php]: Do we need this filter if we're ignoring the pathname above when producing the original set?

File: features/core-verify-checksums.feature [hunk 2]
```diff
@@ -29,6 +30,7 @@ Feature: Validate checksums for WordPress install
     Then STDERR should be:
       """
       Warning: File doesn't exist: readme.html
+      Warning: File should not exist: readme.html.bak
```
Comment (danielbachhuber) [inline features/core-verify-checksums.feature]: Given that it's relatively standard to include extra files in the web root, I was thinking that we'd exclusively check `wp-admin` and `wp-includes` — and ignore the web root entirely. Thoughts?

Comment (gilbitron) [inline features/core-verify-checksums.feature]: Yes I forgot about that. Will fix.

File: php/commands/core.php (hunk 1 above)
Comment (gilbitron) [inline php/commands/core.php]: In the foreach https://github.com/wp-cli/wp-cli/pull/2638/files#diff-97e24ceb3b4ab63c888f11417f6f356fR847 you're skipping the `wp-content` folder, but they still exist in the `$checksums` array.

File: features/core-verify-checksums.feature [hunk 3]
```diff
@@ -63,3 +63,19 @@ Feature: Validate checksums for WordPress install
...
       Success: WordPress install verifies against checksums.
       """
+
+  Scenario: Verify core checksums with extra files
+    Given a WP install
+
+    When I run `wp core update`
+    Then STDOUT should not be empty
+
+    When I run `touch wp-includes/extra-file.txt`
```
Comment (gilbitron) [inline features/core-verify-checksums.feature]: @danielbachhuber Not sure why this isn't working. Any idea how to create an extra file to test this with?

Comment (danielbachhuber) [inline features/core-verify-checksums.feature]: > Any idea how to create an extra file to test this with?

```
Given a wp-includes/extra-file.txt file:
  """
  hello world
  """
```

File: features/core-verify-checksums.feature [hunk 4]
```diff
@@ -63,3 +63,22 @@ Feature: Validate checksums for WordPress install
...
       Success: WordPress install verifies against checksums.
       """
+
+  Scenario: Verify core checksums with extra files
+    Given a WP install
+
+    When I run `wp core update`
+    Then STDOUT should not be empty
+
+    Given a wp-includes/extra-file.txt file:
```
Comment (gilbitron) [inline features/core-verify-checksums.feature]: @danielbachhuber Still can't get this to work?

File: features/core-verify-checksums.feature [hunk 5]
```diff
@@ -63,3 +63,22 @@ Feature: Validate checksums for WordPress install
...
+      """
+      hello world
+      """
+    Then the wp-includes/extra-file.txt file should exist
+
+    When I run `wp core verify-checksums`
+    Then STDOUT should be:
+      """
+      Warning: File should not exist: wp-includes/extra-file.txt
```
Comment (danielbachhuber) [inline features/core-verify-checksums.feature]: This warning is in `STDERR`.

```
Then STDERR should be:
  """
  Warning: File should not exist: wp-includes/extra-file.txt
  """
And STDOUT should be:
  """
  Success: WordPress install verifies against checksums.
  """
```

Comment (gilbitron) [inline features/core-verify-checksums.feature]: That's the one :+1: Thanks.

**Comment by danielbachhuber** [comment]
🚢 🚢 thanks!

**Comment by anttiviljami** [comment]
Thanks, @gilbitron. Needed this! :)

**Pull request merged**